File: ruby/testcode/loginjection_047.rb
require_relative 'shared'
require 'logger'

# vuln-code-snippet start ruby_loginj_progname_constant
def log_progname_constant(req)
  Logger.new($stdout, progname: 'MyApp').info("request processed") # vuln-code-snippet safe-line ruby_loginj_progname_constant
  BenchmarkResponse.json({ ok: true })
end
# vuln-code-snippet end ruby_loginj_progname_constant

File: rust/testcode/loginjection_047.rs
//! CWE-117: User input unconditionally replaced before reaching log call.

// vuln-code-snippet start testcodeLoginjection047
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let mut username = req.param("username");
    username = "redacted".to_string();
    log_info(&format!("Login attempt: user={}", username)); // vuln-code-snippet target-line testcodeLoginjection047
    super::shared::BenchmarkResponse::ok("Logged")
}

fn log_info(msg: &str) {
    eprintln!("[INFO] {}", msg);
}
// vuln-code-snippet end testcodeLoginjection047
